src/main/scala/db/migration/V14__CreateMissingFilePaths.scala

```diff
@@ -0,0 +1,93 @@
+package db.migration
+
+import org.flywaydb.core.api.migration.{BaseJavaMigration, Context}
+import org.json4s.JsonAST.{JArray, JValue}
+import org.json4s.{JNothing, JObject, JString, JValue}
+import org.json4s.native.JsonMethods.{compact, parse, render}
+import org.json4s.native.Serialization
+import org.postgresql.util.PGobject
+import scalikejdbc._
+
+class V14__CreateMissingFilePaths extends BaseJavaMigration {
+  implicit val formats = org.json4s.DefaultFormats
+  case class languageObject(language: String)
+  case class filePathObject(filePath: String, fileSize: Long, language: String, mimeType: String)
+
+  override def migrate(context: Context): Unit = {
+    val db = DB(context.getConnection)
+    db.autoClose(false)
+
+    db.withinTx { implicit session =>
+      allAudios.map {
+        case (id: Long, document: String) => update(convertDocument(document), id)
+      }
+    }
+  }
+
+  def allAudios(implicit session: DBSession): List[(Long, String)] = {
+    sql"select id, document from audiodata"
+      .map(rs => (rs.long("id"), rs.string("document")))
+      .list()
+      .apply()
+  }
+
+  def convertDocument(document: String): String = {
+
+    val oldArticle = parse(document)
+    val newArticle = oldArticle.mapField {
+      case ("filePaths", filePaths: JArray) =>
+        "filePaths" -> {
+          if (filePaths.children.length.equals(0)) {
+            JArray(List.empty[JValue])
+          } else {
+            JArray({
+              val supportedLanguages = ((oldArticle \ "tags") ++ (oldArticle \ "titles") ++ (oldArticle \ "filePaths"))
+                .extract[List[languageObject]]
+                .map(f => f.language)
+                .distinct
+
+              supportedLanguages.map(supportedLang => {
+                val existingFilePath = filePaths.children.find(filePath => {
+                  val lang = filePath
+                    .findField(field => {
+                      field._1.equals("language")
+                    })
+                    .get
+                    ._2
+                    .asInstanceOf[JString]
+
+                  lang.s.equals(supportedLang)
+
+                })
+
+                existingFilePath match {
+                  case Some(filePath) => {
+                    filePath
+                  }
+                  case None =>
+                    val fileObjects = filePaths.extract[List[filePathObject]]
+                    val newFilePath = fileObjects.find(fp => fp.language.equals("nb")).getOrElse(fileObjects.head)
+                    parse(Serialization.write(newFilePath.copy(language = supportedLang)))
+
+                }
+              })
+            })
+          }
+        }
+
```

Comment:
Er ikke eeegentlig verdens største problem i en migrering, men jeg syns denne var liiitt vanskelig å lese, og jeg elsker ikke muligheten for exceptions (`.get` på `Option`, og castinga med `asInstanceOf`.)

Så jeg lagde et lite forslag:
`findLanguagePrioritized` funksjonen finnes egentlig allerede i `Language` funksjonen, men tenkte den var grei nok å lage på nytt for å slippe at migreringen må oppdateres dersom funksjonen endrer seg i fremtiden.
Også renamet jeg klassene dine til å starte med stor bokstav siden vi har det sånn andre plasser :smile: 

```suggestion
  def findLanguagePrioritized[P <: { def language: String }](sequence: Seq[P], language: String): Option[P] = {
    sequence
      .find(_.language == language)
      .orElse(sequence.sortBy(lf => languagePriority.reverse.indexOf(lf.language)).lastOption)
  }

  val languagePriority = List(
    "nb",
    "nn",
    "unknown",
    "en",
    "fr",
    "de",
    "se",
    "sma",
    "es",
    "zh"
  )
  
  def convertDocument(document: String): String = {
    val oldArticle = parse(document)
    val newArticle = oldArticle.mapField {
      case ("filePaths", filePaths: JArray) =>
        val oldFilePaths = filePaths.extract[List[FilePathObject]]
        val supportedLanguages = ((oldArticle \ "tags") ++ (oldArticle \ "titles") ++ (oldArticle \ "filePaths"))
          .extract[List[LanguageObject]]
          .map(_.language)
          .distinct

        val newFilePaths = supportedLanguages.flatMap(lang => {
          findLanguagePrioritized(oldFilePaths, lang)
            .map(_.copy(language = lang))
        })

        "filePaths" -> Extraction.decompose(newFilePaths)
      case x => x

    }

    compact(render(newArticle))
  }

```